packages/components/src/notice/index.js

```diff
@@ -8,22 +8,70 @@ import classnames from 'classnames';
  * WordPress dependencies
  */
 import { __ } from '@wordpress/i18n';
-import { RawHTML } from '@wordpress/element';
+import { RawHTML, useEffect, renderToString } from '@wordpress/element';
+import { speak } from '@wordpress/a11y';
 
 /**
  * Internal dependencies
  */
 import { Button, IconButton } from '../';
 
+/**
+ * Custom hook which announces the message with the given politeness, if a
+ * valid message is provided.
+ *
+ * @param {?string} message    Message to announce.
+ * @param {string}  politeness Politeness in which to announce, one of 'polite'
+ *                             or 'assertive'.
+ */
+function useSpokenMessage( message, politeness ) {
+	useEffect( () => {
+		if ( ! message ) {
+			return;
+		}
+
+		if ( typeof message !== 'string' ) {
+			message = renderToString( message );
+		}
+
+		speak( message, politeness );
+	}, [ message, politeness ] );
+}
+
+/**
+ * Given a notice status, returns an assumed default equivalent role for the
+ * status. Defaults to 'alert'.
+ *
+ * @param {?string} status Notice status.
+ *
+ * @return {string} Notice role.
+ */
+function getDefaultRole( status ) {
+	switch ( status ) {
+		case 'success':
+		case 'warning':
+		case 'info':
+			return 'status';
+
+		case 'error':
+		default:
+			return 'alert';
+	}
+}
+
 function Notice( {
 	className,
-	status,
+	status = 'info',
 	children,
 	onRemove = noop,
 	isDismissible = true,
 	actions = [],
+	role = getDefaultRole( status ),
 	__unstableHTML,
 } ) {
+	const politeness = ( role === 'status' ? 'polite' : 'assertive' );
+	useSpokenMessage( children, politeness );
```

Comment:
I thought we had a way to override the message using a custom notice property? (instead of `children`)

Reply:
> I thought we had a way to override the message using a custom notice property? (instead of `children`)

Yes, I realized after that this isn't quite right. We'll still want some defaults to ensure that ad hoc usage of the `Notice` component announces something with sensible defaulting to `children`, but also deferring to the notice state `spokenMessage` if and as provided.

Reply:
> I thought we had a way to override the message using a custom notice property? (instead of `children`)

Technically, this feature was never implemented for the notices state. It was also a factor in the insistence that the `content` of a notice be a plain string. The value of `speak` has always been configurable, but purely as a boolean value of controlling whether the message is spoken or not (defaulting to `true`).

I can see how we should want to support custom messages. It's not clear to me that this should be done here, however. It might be worth thinking through how that would look as its own dedicated changeset, considering whether to overload the `speak` option for this purpose (allow a string value to be used as the custom spoken message) or a separate option for `spokenMessage` dedicated to this option.

Reply:
The changes as presented here currently set us up pretty well to be able to support this as well, considering that the new state shape of the notice object includes a new `spokenMessage` property.